packages/common/src/content/fetch.ts

```diff
@@ -24,21 +27,42 @@ export interface IFetchContentOptions extends IFetchItemAndEnrichmentsOptions {
   siteOrgKey?: string;
 }
 
+const maybeFetchLayerEnrichments = async (
+  itemAndEnrichments: IItemAndEnrichments,
+  options?: IFetchContentOptions
+) => {
+  // determine if this is a client-side feature layer view
+  const { item, data, layers } = itemAndEnrichments;
+  const layer =
+    layers && getItemLayer(item, layers, options && options.layerId);
+  // TODO: add recordCount here too?
+  const layerEnrichments =
+    layer && isLayerView(layer) && !data
+      ? // NOTE: I'm not sure what conditions causes a layer view
+        // to store (at least part of) it's view definition in item data
+        // it seems that most do not, but until we have a reliable signal
+        // we just fetch the item data for all layer views
```

Comment:
Furthermore, spot checking a handful of such items from this:  
https://qaext.arcgis.com/sharing/rest/search?num=100&start=0&f=json&q=typekeywords%3A%22View%20Service%22
showed that very few had their definition stored in item data, ex:
https://qaext.arcgis.com/sharing/rest/content/items/864c9eaf76ab47ba93d7308bacf1a34f/data?f=json, so most of the time we're fetching the data for nothing.